Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/AzureCliAzureManager.java

```diff
@@ -0,0 +1,253 @@
+            try {
+                getInstance().signIn();
+            } catch (AzureExecutionException ignore) {
+                // Catch the exception when restore
+            } finally {
+                return authMethodDetails;
+            }
+        }
+    }
+
+    public static AzureCliAzureManager getInstance() {
+        return LazyLoader.INSTANCE;
+    }
+
+    @Override
+    public Azure getAzure(String sid) throws IOException {
+        if (!isSignedIn()) {
+            return null;
+        }
+        if (sidToAzureMap.containsKey(sid)) {
+            return sidToAzureMap.get(sid);
+        }
+        final Azure azure = auth().withSubscription(sid);
+        // TODO: remove this call after Azure SDK properly implements handling of unregistered provider namespaces
+        AzureRegisterProviderNamespaces.registerAzureNamespaces(azure);
+        sidToAzureMap.put(sid, azure);
+        return azure;
+    }
+
+    @Override
+    public AppPlatformManager getAzureSpringCloudClient(String sid) {
+        return isSignedIn() ? sidToAzureSpringCloudManagerMap.computeIfAbsent(sid, s ->
+                buildAzureManager(AppPlatformManager.configure()).authenticate(azureCliCredentials, s)) : null;
+    }
+
+    @Override
+    public InsightsManager getInsightsManager(String sid) {
+        return isSignedIn() ? sidToInsightsManagerMap.computeIfAbsent(sid, s ->
+                buildAzureManager(InsightsManager.configure()).authenticate(azureCliCredentials, s)) : null;
+    }
+
+    @Override
+    public List<Subscription> getSubscriptions() {
+        return isSignedIn() ? authenticated.subscriptions().list() : Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public List<Pair<Subscription, Tenant>> getSubscriptionsWithTenant() {
+        if (!isSignedIn()) {
+            return Collections.EMPTY_LIST;
+        }
+        final Tenant subscriptionTenant = authenticated.tenants().list().stream()
+                .filter(tenant -> StringUtils.equals(tenant.tenantId(), authenticated.tenantId()))
+                .findFirst().orElse(null);
+        if (subscriptionTenant == null) {
+            return Collections.EMPTY_LIST;
+        }
+        final List<Pair<Subscription, Tenant>> result = new ArrayList<>();
+        for (Subscription subscription : getSubscriptions()) {
+            result.add(new Pair<>(subscription, subscriptionTenant));
+        }
+        return result;
+    }
+
+    @Override
+    public Settings getSettings() {
+        return settings;
+    }
+
+    @Override
+    public SubscriptionManager getSubscriptionManager() {
+        return subscriptionManager;
+    }
+
+    @Override
+    public void drop() throws IOException {
+        authenticated = null;
+        azureCliCredentials = null;
+        subscriptionManager.cleanSubscriptions();
+    }
+
+    @Override
+    public KeyVaultClient getKeyVaultClient(String tid) {
+        if (!isSignedIn()) {
```

Comment:
use the unified style like spring and ai

Reply:
Will fixed in later pr together with access token/service principal manager